osu.Game.Tests/Visual/Playlists/TestSceneAddPlaylistToCollectionButton.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Audio;
+using osu.Framework.Graphics;
+using osu.Framework.Platform;
+using osu.Framework.Testing;
+using osu.Game.Beatmaps;
+using osu.Game.Database;
+using osu.Game.Online.Rooms;
+using osu.Game.Overlays;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Osu;
+using osu.Game.Screens.OnlinePlay.Playlists;
+using osuTK;
+
+namespace osu.Game.Tests.Visual.Playlists
+{
+    public partial class TestSceneAddPlaylistToCollectionButton : OsuTestScene
+    {
+        private BeatmapManager manager = null!;
+        private BeatmapSetInfo importedBeatmap = null!;
+        private Room room = null!;
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host, AudioManager audio)
+        {
+            Dependencies.Cache(new RealmRulesetStore(Realm));
+            Dependencies.Cache(manager = new BeatmapManager(LocalStorage, Realm, API, audio, Resources, host, Beatmap.Default));
+            Dependencies.Cache(Realm);
+        }
+
+        [Cached(typeof(INotificationOverlay))]
+        private NotificationOverlay notificationOverlay = new NotificationOverlay
+        {
+            Anchor = Anchor.TopRight,
+            Origin = Anchor.TopRight,
+        };
+
+        [SetUpSteps]
+        public void SetUpSteps()
+        {
+            importBeatmap();
+
+            setupRoom();
+
+            AddStep("create button", () =>
+            {
+                AddRange(new Drawable[]
+                {
+                    notificationOverlay,
+                    new AddPlaylistToCollectionButton(room)
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Size = new Vector2(300, 40),
+                    }
+                });
+            });
+        }
```

Comment:
At the point you've got the test this far, you could probably click the button and check the collection is created.